src/bluesky/tests/test_tracing.py

```diff
@@ -1,11 +1,31 @@
 from collections.abc import Generator
 
+import pytest
 from opentelemetry.trace import get_current_span
 
+from bluesky import plans as bp
 from bluesky.plan_stubs import sleep
+from bluesky.tests import requires_ophyd
 from bluesky.tracing import trace_plan, tracer
 
 
+@pytest.fixture(scope="session")
+def with_otl_instrumentation():
+    from opentelemetry import trace
```

Comment:
```suggestion
    pytest.importorskip('opentelemetry')
    from opentelemetry import trace
```

This will make the tests fail gracefully if the dependency is missing.  Even though it is now in the `[dev]` group, it is still better to have the tests gracefully skip on optional dependencies than to hard-fail at test collection.